pennylane/operation.py

```diff
@@ -878,30 +878,34 @@ def __init__(self, *params, wires=None, do_queue=True, id=None):
         self._id = id
         self.queue_idx = None  #: int, None: index of the Operator in the circuit queue, or None if not in a queue
 
+        wires_from_args = False
         if wires is None:
-            raise ValueError(f"Must specify the wires that {self.name} acts on")
+            try:
+                wires = params[-1]
+                params = params[:-1]
+                wires_from_args = True
+            except IndexError as err:
```

Comment:
Would this ever explicitly raise an `IndexError` ? I thought `params[-1]` would always return the last item in the list? When would we cause the `IndexError`

Reply:
If nothing at all is passed.  For example, someone tries `qml.S()`. Just copied over this behavior from the initialization of `Observable`.